doc/api/assert.md

```diff
@@ -1164,6 +1164,9 @@ changes:
 Tests strict equality between the `actual` and `expected` parameters as
 determined by the [SameValue Comparison][].
 
+[`assert.deepStrictEqual()`][] should be used for all objects, not only
+nested ones.
+
```

Comment:
If we do wish to include some text along these lines it should go at the end of the entry, or at least after the sample code.

Reply:
Sure! Once there is a consensus on the content / wording of the hint, I'll push a commit that moves it to the end of the entry.